dev/tools/gen_defaults/lib/template.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+abstract class TokenTemplate {
+  TokenTemplate(this.fileName, this.tokens);
+
+  static const String beginGeneratedComment = '''
+
+// BEGIN GENERATED TOKEN PROPERTIES
+''';
+
+  static const String headerComment = '''
+
+// Generated code to the end of this file. Do not edit by hand.
+// These defaults are generated from the Material Design Token
+// database by the script dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+''';
+
+  static const String endGeneratedComment = '''
+
+// END GENERATED TOKEN PROPERTIES
+''';
+
+  final String fileName;
+  final Map<String, dynamic> tokens;
+
+  /// Replace or append the contents of the file with the text from [generate].
+  ///
+  /// If the file already contains generated block at the end, it will
+  /// be replaced by the [generate] output. Otherwise the content will
+  /// just be appended to the end of the file.
+  Future<void> updateFile() async {
+    String contents = File(fileName).readAsStringSync();
+    final int previousGeneratedIndex = contents.indexOf(beginGeneratedComment);
+    if (previousGeneratedIndex != -1) {
+      contents = contents.substring(0, previousGeneratedIndex);
+    }
+    final StringBuffer buffer = StringBuffer(contents);
+    buffer.write(beginGeneratedComment);
+    buffer.write(headerComment);
+    buffer.write(generate());
+    buffer.write(endGeneratedComment);
+    File(fileName).writeAsStringSync(buffer.toString());
+  }
+
+  /// Provide the generated content for the template.
+  ///
+  /// This abstract method needs to be implemented by subclasses
+  /// to provide the content that [updateFile] will append to the
+  /// bottom of the file.
+  String generate();
+
+  String color(String tokenName) {
```

Comment:
We could do all kinds of validity checking here (and elsewhere) so that the tool would fail more gracefully and informatively if the token database JSON was corrupted. Since this is just a tool we'll run as-needed and by-hand, it's fine to put off that kind of work until it seems like we need it.

Reply:
Sure thing. This was just a first pass. I should probably document these helper methods a bit better as well. I will add it to my list of possible improvements. Thx.